internal/utilities/utilities.go

```diff
@@ -42,3 +47,41 @@ func MergeMaps(maps ...map[string]string) map[string]string {
 func AddTenantPrefix(name string, tenantControlPlane *kamajiv1alpha1.TenantControlPlane) string {
 	return fmt.Sprintf("%s%s%s", tenantControlPlane.GetName(), separator, name)
 }
+
+// EncondeToYaml returns the given object in yaml format and the error.
+func EncondeToYaml(o runtime.Object) ([]byte, error) {
+	scheme := runtime.NewScheme()
+	encoder := json.NewYAMLSerializer(json.SimpleMetaFactory{}, scheme, scheme)
+	buf := bytes.NewBuffer([]byte{})
+	err := encoder.Encode(o, buf)
+
+	return buf.Bytes(), err
+}
+
+// IsValidIP checks if the given argument is an IP.
+func IsValidIP(ip string) bool {
+	return IsValidIPV4(ip)
+}
+
+// IsValidIPV4 checks if the given argument is a IP v4.
+func IsValidIPV4(ip string) bool {
+	pattern := "^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$"
+
+	return validateRegex(pattern, ip)
+}
```

Comment:
Why not use `net.ParseIP` here?

Reply:
no idea about that function. I will change